betterproto/tests/test_casing.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+
+from betterproto.casing import camel_case, pascal_case, snake_case
+
+
+@pytest.mark.parametrize(
+    ["value", "expected"],
+    [
+        ("", ""),
+        ("a", "A"),
+        ("foobar", "Foobar"),
+        ("FooBar", "FooBar"),
+        ("foo.bar", "FooBar"),
+        ("foo_bar", "FooBar"),
```

Comment:
Hm yes it does create collissions. However, I do not see a practical way to avoid collissions and still convert all other cases to PascalCase, since its not guaranteed the that source strings are even snake_case.

For example:
```protobuf
message MyMessage {

}
```
should generate `MyMessage`

and
```protobuf
message my_message {

}
```
should generate `MyMessage` as well.

If we generate for the last case `My_Message`, we are not really converting anything to PascalCase, and we might as well use the original name.

The same problem goes for the fields.
I think we are generating snake_case there to make the code look nice and pythonic, like we are generating PascalCase for classes to make them nicely pythonic.

In my opinion, two messages or fields with names that are only distinguishable by casing style, is not really a valid use case, but maybe I'm biased..

Reply:
The argument to be made here is that for users following best practices of message naming it doesn't matter. For those who are, it might matter and they might not have a choice, because they might be consuming an API they don't control.